### PR TITLE
Feature/improve blockchain adapter api library

### PR DESCRIPTION
**Pull request by jbern0rd**

### CODE REVIEW

**Comment by mcornaton** [comment]
Do you think there could be some sort of hint on which APIs are authenticated and which ones are not? Something like `// region Authenticated APIs` comment?

File: iexec-blockchain-adapter-api-library/src/main/java/com/iexec/blockchain/api/BlockchainAdapterApiClientBuilder.java [hunk 1]
```diff
@@ -2,21 +2,36 @@
...
-                        username,
-                        password
-                );
+    /**
+     * Create an unauthenticated feign client to query apis described in {@link BlockchainAdapterApiClient}.
+     * @param logLevel Feign logging level to configure.
+     * @param url Url targeted by the client.
+     * @return Feign client for {@link BlockchainAdapterApiClient} apis.
+     */
+    public static BlockchainAdapterApiClient getInstance(Logger.Level logLevel, String url) {
```
Comment (zguesmi) [inline iexec-blockchain-adapter-api-library/src/main/java/com/iexec/blockchain/api/BlockchainAdapterApiClientBuilder.java:19]: In which case do we need the unauthenticated client?

Comment (jbern0rd) [reply iexec-blockchain-adapter-api-library/src/main/java/com/iexec/blockchain/api/BlockchainAdapterApiClientBuilder.java:19]: We need it to access public chain config from worker which is unauthenticated.

**Comment by jbern0rd** [comment]
> Do you think there could be some sort of hint on which APIs are authenticated and which ones are not? Something like `// region Authenticated APIs` comment?

Done in d4d80e

File: iexec-blockchain-adapter-api-library/src/main/java/com/iexec/blockchain/api/BlockchainAdapterApiClientBuilder.java [hunk 2]
```diff
@@ -2,21 +2,36 @@
...
-                                                         String url, String username, String password) {
-        BasicAuthRequestInterceptor requestInterceptor =
-                new BasicAuthRequestInterceptor(
-                        username,
-                        password
-                );
+    /**
+     * Create an unauthenticated feign client to query apis described in {@link BlockchainAdapterApiClient}.
+     * @param logLevel Feign logging level to configure.
+     * @param url Url targeted by the client.
```
Comment (mcornaton) [inline iexec-blockchain-adapter-api-library/src/main/java/com/iexec/blockchain/api/BlockchainAdapterApiClientBuilder.java:16]: Shouldn't that `url` parameter be renamed into `baseUrl`?

Comment (jbern0rd) [reply iexec-blockchain-adapter-api-library/src/main/java/com/iexec/blockchain/api/BlockchainAdapterApiClientBuilder.java:16]: I used the same parameter name as the `target` method from feign API.
I think this is good enough.

**Comment by jbern0rd** [comment]
> Do you think there could be some sort of hint on which APIs are authenticated and which ones are not? Something like `// region Authenticated APIs` comment?

should be dealt with in 5764b36

**Review by mcornaton — APPROVED** [review]
Thanks!

**Pull request merged**